packages/core/status/core-status-server-internal/src/log_plugins_status.ts

```diff
@@ -0,0 +1,187 @@
+      const allAvailableSuffix =
+        status.available.length === total ? ` (all ${total} plugins are now available)` : '';
+
+      if (inferredCount === 0) {
+        messages.push(
+          `'${statusReportingPlugins[0].pluginName}' is now ${currentLevel}: ${reason}${allAvailableSuffix}`
+        );
+      } else {
+        messages.push(
+          `'${statusReportingPlugins[0].pluginName}' (and ${inferredCount} more) are now ${currentLevel}: ${reason}${allAvailableSuffix}`
+        );
+      }
+    } else if (statusReportingPlugins.length > 1) {
+      const multilineStatus = [];
+      if (inferredCount === 0) {
+        multilineStatus.push(
+          `${
+            statusReportingPlugins.length
+          } plugins are now ${currentLevel}: ${statusReportingPlugins
+            .map(({ pluginName }) => pluginName)
+            .join(', ')}`
+        );
+      } else {
+        multilineStatus.push(
+          `The following plugins are now ${currentLevel}: ${statusReportingPlugins
+            .map(({ pluginName }) => pluginName)
+            .join(', ')} (and ${inferredCount} more)`
+        );
+      }
+
+      statusReportingPlugins.forEach(({ pluginName, ...pluginStatus }) =>
+        multilineStatus.push(
+          ` - '${pluginName}' is now ${currentLevel}: ${getReason(pluginStatus)}`
+        )
+      );
+
+      messages.push(multilineStatus.join('\n'));
+    }
+  });
+
+  // report plugins that are STILL unhealthy (even though their status has not changed)
+  Object.entries(status)
+    .filter(([currentLevel]) => currentLevel !== ServiceStatusLevels.available.toString())
+    .forEach(([currentLevel, pluginStatuses]) => {
+      const statusReportingPlugins = pluginStatuses
+        .filter(
+          (pluginStatus) =>
+            !updates[currentLevel as PluginLevel].find(
+              (updatedStatus) => updatedStatus.pluginName === pluginStatus.pluginName
+            )
+        )
+        .filter(({ isReportedStatus }) => isReportedStatus);
+
+      const inferredCount = pluginStatuses.length - statusReportingPlugins.length;
+
+      if (statusReportingPlugins.length === 1) {
+        const reason = getReason(statusReportingPlugins[0]);
+        if (inferredCount === 0) {
+          messages.push(`${statusReportingPlugins[0].pluginName} is ${currentLevel}: ${reason}`);
+        } else {
+          messages.push(
+            `'${statusReportingPlugins[0].pluginName}' plugin (and some others that depend on it) are ${currentLevel}: ${reason}`
+          );
+        }
+      } else if (statusReportingPlugins.length > 1) {
+        if (inferredCount === 0) {
+          messages.push(
+            `The following plugins are ${currentLevel}: ${statusReportingPlugins.map(
+              ({ pluginName }) => pluginName
+            )}`
+          );
+        } else {
+          messages.push(
+            `The following plugins (and some others that depend on them) are ${currentLevel}: ${statusReportingPlugins.map(
```

Comment:
NIT: `some` again, can we display the actual number instead, and would that be useful?

Reply:
> The problem is that I don't have information of what plugins depend on the one that is e.g. degraded.
I only know they are also degraded, but they might depend on other plugins that just switched to degraded.

Yeah but we're displaying a single message for all the degraded plugins (`The following pluginS`), right? so even if we don't know which one(s) caused the "child" plugins to degrade, we still know that it was all caused by one, or more non-inferred plugins?

Reply:
There's a distinction between:
1. _The following plugins are NOW `degraded`_ ... in this scenario I know that the "child" ones have just become degraded, caused by the non-inferred ones.
2. _The following plugins are `degraded`_ ... in this scenario, I know that some plugins are degraded, but I don't know if they were already degraded or they have just become degraded. I could perhaps do a `diff` and remove the newly degraded ones, which will be logged in the other message (1). Or I could simply not log the non-healthy ones that haven't changed status.

Reply:
Okay, then we may want to take some sync time to talk about it, as I thought we were only displaying messages for the first case (when plugin statuses effectively change)